crates/ty_python_semantic/resources/mdtest/typed_dict.md

```diff
@@ -468,10 +470,144 @@ class Employee(TypedDict):
     name: str
     employee_id: int
 
-p1: Person = Employee(name="Alice", employee_id=1)
+class Robot(TypedDict):
+    name: int
 
-# TODO: this should be an error
-e1: Employee = Person(name="Eve")
+static_assert(is_assignable_to(Employee, Person))
+
+static_assert(not is_assignable_to(Person, Employee))
+static_assert(not is_assignable_to(Robot, Person))
+static_assert(not is_assignable_to(Person, Robot))
+```
+
+In order for one `TypedDict` `B` to be assignable to another `TypedDict` `A`, all required keys in
+`A`'s schema must also be required in `B`'s schema. If a key is not-required and also mutable in
+`A`, then it must be not-required in `B` also (because `A` allows the caller to `del` that key).
+These rules cover keys that are explicitly marked `NotRequired`, and also all the keys in a
+`TypedDict` with `total=False`.
+
+```py
+from typing_extensions import NotRequired
+
+class Spy(TypedDict):
+    name: NotRequired[str]
+
+# invalid because `Spy` might be missing `name`
+static_assert(not is_assignable_to(Spy, Person))
+
+# invalid because `Spy` is allowed to delete `name`, while `Person` is not
+static_assert(not is_assignable_to(Person, Spy))
+
+class Amnesiac(TypedDict, total=False):
+    name: ReadOnly[str]
+
+# invalid because `Amnesiac` might be missing `name`
+static_assert(not is_assignable_to(Amnesiac, Person))
+
+# Allowed. `Amnesiac` can't delete `name`, because it's read-only.
+static_assert(is_assignable_to(Person, Amnesiac))
+```
+
+If an item in `A` (the destination `TypedDict` type) is read-only, then the corresponding item in
+`B` can have any assignable type. But if the item in `A` is mutable, the item type in `B` must match
+exactly. The required and not-required cases are different codepaths, so we need test all the
+permutations:
+
+```py
+from typing import Any
```

Comment:
We import `Any` here, but it doesn't seem to be used in this block of tests.

I think we should add some tests with typeddicts with `Any` typed field(s), showing how that impacts subtyping and assignability. Assignability should be forgiving (as shown in the comment above); subtyping should be strict.